sdk/keyvault/Azure.Security.KeyVault.Secrets/src/DeleteSecretOperation.cs

```diff
@@ -5,27 +5,26 @@
 using System.Threading;
 using System.Threading.Tasks;
 using Azure.Core;
-using Azure.Core.Pipeline;
 
 namespace Azure.Security.KeyVault.Secrets
 {
     /// <summary>
     /// A long-running operation for <see cref="SecretClient.StartDeleteSecret(string, CancellationToken)"/> or <see cref="SecretClient.StartDeleteSecretAsync(string, CancellationToken)"/>.
     /// </summary>
-    public class DeleteSecretOperation : Operation<DeletedSecret>
+    public class DeleteSecretOperation : Operation<DeletedSecret>, IOperation
     {
         private static readonly TimeSpan s_defaultPollingInterval = TimeSpan.FromSeconds(2);
 
         private readonly KeyVaultPipeline _pipeline;
+        private readonly OperationInternal _operationInternal;
         private readonly DeletedSecret _value;
-        private Response _response;
-        private bool _completed;
+        private readonly bool _completed;
 
         internal DeleteSecretOperation(KeyVaultPipeline pipeline, Response<DeletedSecret> response)
         {
             _pipeline = pipeline;
             _value = response.Value ?? throw new InvalidOperationException("The response does not contain a value.");
-            _response = response.GetRawResponse();
+            _operationInternal = new(_pipeline.Diagnostics, this, response.GetRawResponse(), nameof(DeleteSecretOperation));
 
             // The recoveryId is only returned if soft-delete is enabled.
             if (_value.RecoveryId is null)
```

Comment:
Can we move this logic into the `IOperation.UpdateStateAsync`? That way you'll be able to avoid having the `_completed` field.

Reply:
Ah, I see, not you can't. Right now HasCompleted is set to true at the time of construction.

Reply:
Wonder if we should add _operationInternal.SetState(OperationState). That way we can set both the value and completeness from the ctor.

Reply:
I must've missed that. I saw `ApplyState` but it was `protected`. But I don't want to complete it in the ctor, only store the value.

Reply:
You didn't miss it. It's not there. But I think it might be a good feature to add.

Reply:
Per offline discussion, I'll add a sync-only `SetState(OperationState state)` and `SetState(OperationState<T> state)` accordingly.